# HG changeset patch
# User Jordon Moss <mossj32@gmail.com>
# Date 1744971351 10800
#      Fri Apr 18 07:15:51 2025 -0300
# Node ID 020695547d33a6335060fa28443bc1058614ee8c
# Parent  30e122fe991b5047090d50583033c0e5ec87d001
# EXP-Topic strikers_stuff
Add APROP_SpawnFlags.

diff -r 30e122fe991b -r 020695547d33 docs/zandronum-history.txt
--- a/docs/zandronum-history.txt	Fri Apr 18 07:04:29 2025 -0300
+++ b/docs/zandronum-history.txt	Fri Apr 18 07:15:51 2025 -0300
@@ -26,6 +26,7 @@
 +	- Backport A_FaceTarget flags from ZDoom revision 53fd57d6, and add a new flag, FAF_PITCHFROMORIGIN for A_FaceTarget. [StrikerTheHedgefox]
 +	- Backport APROP_DamageType. [StrikerTheHedgefox]
 +	- Hitting ice corpses now trigger damage events. [StrikerTheHedgefox]
++	- Added APROP_SpawnFlags, which gets/sets the map spawn flags of an actor. Useful for passing these down to other actors. [StrikerTheHedgefox]
 +	- Added APROP_Special, and APROP_Args0-4 to Get/Set/CheckActorProperty. [StrikerTheHedgefox]
 +	- Added A_JumpTo codepointer, which jumps to a state instantly & with prediction. Useful for jumping to an ancestor actor's state without the issues A_Jump with a 256 chance has. [StrikerTheHedgefox]
 +	- Added "hidelives = true/false" token to SBARINFO, allowing to disable the built-in lives display on the HUD. [StrikerTheHedgefox]
diff -r 30e122fe991b -r 020695547d33 src/p_acs.cpp
--- a/src/p_acs.cpp	Fri Apr 18 07:04:29 2025 -0300
+++ b/src/p_acs.cpp	Fri Apr 18 07:15:51 2025 -0300
@@ -4915,6 +4915,11 @@
 		actor->args[property - APROP_Args0] = value;
 		break;
 
+	case APROP_SpawnFlags: // [JM]
+		actor->SpawnFlags = value;
+		actor->HandleSpawnFlags();
+		break;
+
 	default:
 		// do nothing.
 		break;
@@ -5021,6 +5026,7 @@
 	case APROP_Args2:		return actor->args[2]; // [JM]
 	case APROP_Args3:		return actor->args[3]; // [JM]
 	case APROP_Args4:		return actor->args[4]; // [JM]
+	case APROP_SpawnFlags:	return actor->SpawnFlags; // [JM]
 
 	default:				return 0;
 	}
@@ -5076,6 +5082,9 @@
 		case APROP_Args4: // [JM]
 			return (GetActorProperty(tid, property) == value);
 
+		case APROP_SpawnFlags: // [JM]
+			return !!(GetActorProperty(tid, property) & value); // Special handling to check if bitflag is present.
+
 		// Boolean values need to compare to a binary version of value
 		case APROP_Ambush:
 		case APROP_Invulnerable:
diff -r 30e122fe991b -r 020695547d33 src/p_acs.h
--- a/src/p_acs.h	Fri Apr 18 07:04:29 2025 -0300
+++ b/src/p_acs.h	Fri Apr 18 07:15:51 2025 -0300
@@ -430,6 +430,7 @@
 	APROP_Args2			= 203, // [JM]
 	APROP_Args3			= 204, // [JM]
 	APROP_Args4			= 205, // [JM]
+	APROP_SpawnFlags	= 206, // [JM]
 };	
 
 // [Dusk] Enumeration for GetTeamProperty
